src/System.Xml.XmlSerializer/src/System/Xml/BinHexEncoder.cs

```diff
@@ -11,23 +13,11 @@ internal abstract class BinHexEncoder
 
         internal static string Encode(byte[] inArray, int offsetIn, int count)
         {
-            if (null == inArray)
-            {
-                throw new ArgumentNullException(nameof(inArray));
-            }
-            if (0 > offsetIn)
-            {
-                throw new ArgumentOutOfRangeException(nameof(offsetIn));
-            }
-            if (0 > count)
-            {
-                throw new ArgumentOutOfRangeException(nameof(count));
-            }
-            if (count > inArray.Length - offsetIn)
-            {
-                throw new ArgumentOutOfRangeException(nameof(count));
-            }
-
+            Debug.Assert(null != inArray);
+            Debug.Assert(0 <= offsetIn);
+            Debug.Assert(0 <= count);
+            Debug.Assert(count <= inArray.Length - offsetIn);
```

Comment:
I assume this change is because these arguments are all validated in all of the paths that call to this method?